### PR TITLE
🎨[Code Quality] Refactor /src folder

### PR DESCRIPTION
**Pull request by beefchimi**
This PR refactors the `/src` folder for better organization and code formatting.

I've also included my fix from: https://github.com/streamich/use-media/pull/53

cc @streamich

### CODE REVIEW

File: src/utilities/camel-to-hypen.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export default function camelToHyphen(camelString: string) {
```
Comment (streamich) [inline src/utilities/camel-to-hypen.ts]: I guess it should be "hyphen" with "h".

File: src/index.ts [hunk 2]
```diff
@@ -1,58 +1,2 @@
...
-      mounted = false;
-      mql.removeListener(onChange);
-    };
-  }, [query]);
-
-  return state;
-};
-
-
-export const useMedia = createUseMedia(useEffect);
```
Comment (streamich) [inline src/index.ts]: If we don't export `useMedia`, it is a breaking change. As we are not adding any functionality, but refactoring code, this probably should not be a breaking change.

File: src/index.ts [hunk 3]
```diff
@@ -1,58 +1,2 @@
...
-export const useMediaLayout = createUseMedia(useLayoutEffect);
-
-export default useMedia;
+export {default, useMediaLayout, mockMediaQueryList} from './useMedia';
+export {Effect, MediaQueryObject} from './types';
```
Comment (streamich) [inline src/index.ts]: What is the use case for exporting `mockMediaQueryList`, `Effect` and `MediaQueryObject`? Maybe we can export just the hooks from the top level here? And if somebody wants to import `mockMediaQueryList`, for example, they can still import from `useMedia.ts` file directly.

File: src/useMedia.ts [hunk 4]
```diff
@@ -0,0 +1,53 @@
+import {useState, useEffect, useLayoutEffect} from 'react';
```
Comment (streamich) [inline src/useMedia.ts]: You use camel case naming this file `useMedia.ts`, but in other files you use kebab case, like `camel-to-hypen.ts`.

File: src/utilities/camel-to-hypen.ts (hunk 1 above)
Comment (beefchimi) [reply src/utilities/camel-to-hypen.ts]: ah shoot nice catch!

File: src/useMedia.ts (hunk 4 above)
Comment (beefchimi) [reply src/useMedia.ts]: Ah yea thats just a convention I follow in the projects I work on - utilities are kebab cased... but I don't need to put those conventions onto this codebase. I will rename these utilities to camelCase.

File: src/index.ts (hunk 3 above)
Comment (beefchimi) [reply src/index.ts]: I have reverted back to not exporting the `mockMediaQueryList` + `types` at this entry point.

I have also updated the README to show a use case _(in my app at least)_ for using the `mockMediaQueryList`.

Let me know if you are okay with the latest commits.

**Comment by beefchimi** [comment]
@streamich I've pushed some new commits to address your feedback.

The last commit updates the README to:
1. Highlight the additional exports that are available in the `/lib/` path.
2. Show a test env example.
3. Suggest considering using `Context` to further improve performance.

I can drop the README commit if you think it is unnecessary.

Also, I've tested this branch's `build` output in my project, and everything worked perfectly! Here is a quick screenshot of what the `/lib` folder now looks like:

lib-build

**Pull request merged**